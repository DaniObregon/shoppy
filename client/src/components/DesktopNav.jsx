import React from "react";
import {
  Box,
  Stack,
  Link,
  Popover,
  PopoverTrigger,
  PopoverContent,
  useColorModeValue,
} from "@chakra-ui/react";
import { useSelector } from "react-redux";
import { useNavigate } from "react-router-dom";
import { DesktopSubNav } from "./DesktopSubNav";
import { NAV_ITEMS } from "../utils/navItems";

export const DesktopNav = () => {
  const linkColor = useColorModeValue("gray.600", "gray.200");
  const linkHoverColor = useColorModeValue("gray.800", "white");
  const popoverContentBgColor = useColorModeValue("white", "gray.800");

  // Obtener el rol del usuario desde Redux
  const { role_id } = useSelector((state) => state.userInfo);
  const navigate = useNavigate();

  const handleNavigation = (path) => {
    navigate(path);
  };

  return (
    <Stack direction={"row"} spacing={4}>
      {NAV_ITEMS(role_id).map((navItem) => (
        <Box key={navItem.label}>
          <Popover trigger={"hover"} placement={"bottom-start"}>
            <PopoverTrigger>
              <Link
                p={2}
                onClick={() => handleNavigation(navItem.path)}
                fontSize={"m"}
                fontWeight={500}
                color={linkColor}
                _hover={{
                  textDecoration: "none",
                  color: linkHoverColor,
                }}
              >
                {navItem.label}
              </Link>
            </PopoverTrigger>

            {navItem.children && (
              <PopoverContent
                border={0}
                boxShadow={"xl"}
                bg={popoverContentBgColor}
                p={4}
                rounded={"xl"}
                minW={"sm"}
              >
                <Stack>
                  {navItem.children.map((child) => (
                    <DesktopSubNav key={child.label} {...child} />
                  ))}
                </Stack>
              </PopoverContent>
            )}
          </Popover>
        </Box>
      ))}
    </Stack>
  );
};
