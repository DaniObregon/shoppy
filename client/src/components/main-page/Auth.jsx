import api from "../../config/axiosConfig"
import React from "react";
import { signInWithPopup, GoogleAuthProvider, signOut } from "firebase/auth";
import { SignUpButton } from "./SignUpButton";
import { SignOutButton } from "./SignOutButton";
import { useNavigate } from "react-router-dom";
import { auth } from "../../config/firebaseConfig";
import { useDispatch, useSelector } from "react-redux";
import { login, logout } from "../../redux/slices/userSlice";
import { Flex, Text, Stack } from "@chakra-ui/react";

export const Auth = ({ onAuthSuccess }) => {
  const provider = new GoogleAuthProvider();
  const navigate = useNavigate();
  const dispatch = useDispatch();

  // Obtiene el estado del usuario desde Redux (obtiene los datos)
  const userInfo = useSelector((state) => state.userInfo);

  const signUpWithGoogle = async () => {
    try {
      const result = await signInWithPopup(auth, provider);
      const user = result.user;
      const token = await user.getIdToken();
      const userData = {
        name: user.displayName,
        email: user.email,
        role_id: 1, // Rol por defecto (client)
        token: token, // Guardamos el token aquí
      };

      //console.log("Datos que se envían al backend:", userData);

      const response = await api.post(
        "/api/auth/signup",
        JSON.stringify(userData),
        {
          headers: {
            "Content-Type": "application/json",
            Authorization: `Bearer ${token}`, // El token se envia en un header
          },
        }
      );

      //console.log("Respuesta del backend:", response.data);

      // Accede a las propiedades del objeto `user` desde la respuesta del backend
      const { name, email, role } = response.data.user;

      //console.log("Datos recibidos del backend:", { name: name, email: email, role: role });

      // Actualizar el estado global de Redux con los datos del usuario incluyendo el token
      dispatch(
        login({
          name: name,
          email: email,
          role_id: role,
          token: token,
        })
      );

      // Si la autenticación y la creación del usuario son exitosas
      if (onAuthSuccess) {
        onAuthSuccess(user); // Llamar a la función si está definida
      }
      navigate("/"); // Redirige al home después de autenticarse (o cambiar ruta)
    } catch (error) {
      console.error("Error durante el inicio de sesión:", error);
    }
  };

  const signOutUser = async () => {
    try {
      await signOut(auth);
      dispatch(logout()); // Despachar la acción de logout en Redux
      navigate("/");
    } catch (error) {
      console.error("Error al cerrar sesión:", error);
    }
  };

  return (
    <Flex align="center" justify="center" p={4}>
      {userInfo.name ? (
        <Stack direction="row" align="center" spacing={4}>
          <Text
            p={2}
            color="gray.600"
            whiteSpace="nowrap"
            maxWidth="200px"
            overflow="hidden"
            textOverflow="ellipsis"
            fontSize={"m"}
          >
            Hola, {userInfo.name.split(" ")[0]}
          </Text>
          <SignOutButton onClick={signOutUser} />
        </Stack>
      ) : (
        <SignUpButton onClick={signUpWithGoogle} />
      )}
    </Flex>
  );
};
