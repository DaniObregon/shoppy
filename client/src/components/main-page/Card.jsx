import React from "react";
import { useNavigate } from "react-router-dom";
import { chakra, Box, Flex } from "@chakra-ui/react";
import { AddToCartButton } from "./AddToCartButton";

export const Card = ({ id, brand, model, description, price, imgUrl }) => {
  const navigate = useNavigate();

  const handleNavigate = () => {
    navigate(`/product/${id}`);
  };

  return (
    <Flex
      direction="column"
      justifyContent="center"
      alignItems="center"
      w="full"
      mx="auto"
      mb={4}
      transition="transform 0.15s ease-in-out"
      _hover={{
        transform: "scale(1.015)",
        cursor: "pointer",
      }}
      onClick={handleNavigate} // Redirige cuando se hace clic en la card
    >
      <Box
        bg="gray.300"
        h={{ base: 56, md: 64 }} //Imagen m치s grande para mantener la proporci칩n
        w="full"
        rounded="lg"
        shadow="md"
        bgSize="cover"
        bgPos="center"
        style={{
          backgroundImage: `url(${imgUrl})`,
        }}
      ></Box>

      <Box
        w={{ base: "80%", md: "70%" }} // Caja de datos peque침a y centrada
        bg="white"
        _dark={{ bg: "gray.800" }}
        mt={-10} // Superpone la caja de datos en la imagen
        shadow="lg"
        rounded="lg"
        overflow="hidden"
      >
        <chakra.h3
          py={2}
          textAlign="center"
          fontWeight="bold"
          textTransform="uppercase"
          color="gray.800"
          _dark={{ color: "white" }}
          letterSpacing={1}
        >
          {brand}
        </chakra.h3>
        <chakra.p
          py={1}
          textAlign="center"
          fontWeight="medium"
          color="gray.600"
          _dark={{ color: "gray.300" }}
          fontSize="sm"
        >
          {model}
        </chakra.p>
        <chakra.p
          py={1}
          textAlign="center"
          fontWeight="medium"
          color="gray.600"
          _dark={{ color: "gray.300" }}
          fontSize="sm"
          sx={{
            overflow: "hidden",
            display: "-webkit-box",
            WebkitLineClamp: "3",
            WebkitBoxOrient: "vertical",
            wordBreak: "break-word", // Corta palabras largas autom치ticamente
            overflowWrap: "break-word", // Compatibilidad adicional
          }}
        >
          {description}
        </chakra.p>
        <Flex
          alignItems="center"
          justifyContent="space-between"
          py={2}
          px={3}
          bg="gray.200"
          _dark={{ bg: "gray.700" }}
        >
          <chakra.span
            fontWeight="bold"
            color="gray.800"
            _dark={{ color: "gray.200" }}
          >
            ${price}
          </chakra.span>
          <AddToCartButton id={id} />
        </Flex>
      </Box>
    </Flex>
  );
};
