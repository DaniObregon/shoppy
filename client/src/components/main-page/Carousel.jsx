import React, { useEffect, useState } from "react";
import { Box, Flex, Image, Text } from "@chakra-ui/react";
import { slides } from "../../data/carouselSlides";

export const Carousel = () => {
  const [currentSlide, setCurrentSlide] = useState(0);
  const [visibleSlides, setVisibleSlides] = useState(slides);
  const [isPaused, setIsPaused] = useState(false); // Estado para manejar si la transición está pausada
  const [isHovered, setIsHovered] = useState(false); // Estado para manejar si el mouse está sobre el carrusel

  const slidesCount = visibleSlides.length;
  const carouselStyle = {
    transition: "all 1.5s",
    ml: `-${currentSlide * 100}%`,
  };

  const SLIDES_INTERVAL_TIME = 4000;
  const ANIMATION_DIRECTION = "right";

  useEffect(() => {
    const prevSlide = () => {
      setCurrentSlide((s) => (s === 0 ? slidesCount - 1 : s - 1));
    };

    const nextSlide = () => {
      setCurrentSlide((s) => (s === slidesCount - 1 ? 0 : s + 1));
    };

    const automatedSlide = setInterval(() => {
      if (!isPaused) {
        ANIMATION_DIRECTION.toLowerCase() === "left"
          ? prevSlide()
          : nextSlide();
      }
    }, SLIDES_INTERVAL_TIME);

    return () => clearInterval(automatedSlide);
  }, [slidesCount, isPaused]);

  const handleMouseEnter = () => {
    setIsPaused(true);
    setIsHovered(true);
  };

  const handleMouseLeave = () => {
    setIsPaused(false);
    setIsHovered(false);
  };

  // Función para manejar el error de carga de imágenes y eliminar el slide
  const handleImageError = (index) => {
    setVisibleSlides((prevSlides) =>
      prevSlides.filter((_, slideIndex) => slideIndex !== index)
    );
  };

  return (
    <Flex
      id="carousel"
      role="region"
      aria-label="carousel"
      w="full"
      bg="#edf3f8"
      _dark={{ bg: "#3e3e3e" }}
      p={0}
      alignItems="center"
      justifyContent="center"
      onMouseEnter={handleMouseEnter}
      onMouseLeave={handleMouseLeave}
    >
      <Flex w="full" overflow="hidden">
        <Flex pos="relative" h="500px" w="full" {...carouselStyle}>
          {visibleSlides.map((slide, index) => (
            <Box
              key={`slide-${index}`}
              flex="none"
              boxSize="full"
              shadow="md"
              transition="transform 0.3s ease"
              style={{
                transform: isHovered ? "translateY(-5px)" : "translateY(0px)",
              }}
            >
              <Text
                color="white"
                fontSize="xs"
                p="8px 12px"
                pos="absolute"
                top="0"
                whiteSpace="nowrap"
              >
                {index + 1} / {slidesCount}
              </Text>
              <Image
                src={slide.img}
                alt={`Slide ${index + 1}`}
                boxSize="full"
                objectFit="cover"
                style={{
                  opacity: isHovered ? "0.8" : "1",
                  transition: "opacity 0.3s ease",
                }}
                onError={() => handleImageError(index)} // Maneja el error de carga de imagen eliminando el slide
              />
            </Box>
          ))}
        </Flex>
      </Flex>
    </Flex>
  );
};
