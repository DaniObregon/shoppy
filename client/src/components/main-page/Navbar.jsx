import React from "react";
import {
  Box,
  Flex,
  IconButton,
  Stack,
  Collapse,
  useColorModeValue,
  useDisclosure,
} from "@chakra-ui/react";
import { HamburgerIcon, CloseIcon } from "@chakra-ui/icons";
import { Auth } from "./Auth";
import { DarkModeToggle } from "./DarkModeToggle";
import { Logo } from "./Logo";
import { DesktopNav } from "./DesktopNav";
import { MobileNav } from "./MobileNav";

export const Navbar = () => {
  const { isOpen, onToggle } = useDisclosure();

  return (
    <Box mt="60px" marginBottom={1} marginTop={5}>
      <Flex
        bg={useColorModeValue("white", "gray.800")}
        color={useColorModeValue("gray.600", "white")}
        minH={"60px"}
        py={{ base: 2 }}
        px={{ base: 10 }}
        borderBottom={1}
        borderTop={1}
        borderStyle={"solid"}
        borderColor={useColorModeValue("gray.200", "gray.900")}
        align={"center"}
        justify={"space-between"}
        marginTop={8}
        marginBottom={1}
        paddingTop="9px"
        border={-10}
      >
        <Flex
          flex={{ base: 1, md: "auto" }}
          ml={{ base: -2 }}
          display={{ base: "flex", md: "none" }}
        >
          <IconButton
            onClick={onToggle}
            icon={
              isOpen ? <CloseIcon w={3} h={3} /> : <HamburgerIcon w={5} h={5} />
            }
            variant={"ghost"}
            aria-label={"Toggle Navigation"}
          />
        </Flex>
        <Flex flex={{ base: 1 }} justify={{ base: "center", md: "start" }}>
          <Logo />
          <Flex
            display={{ base: "none", md: "flex" }}
            ml={10}
            alignItems={"center"}
          >
            <DesktopNav />
          </Flex>
        </Flex>

        <Stack
          flex={{ base: 1, md: 0 }}
          justify={"flex-end"}
          direction={"row"}
          spacing={6}
          alignItems="center" // Asegura que todo estÃ© alineado verticalmente
        >
          <Auth /> {/* Este componente maneja tanto login como logout */}
          <DarkModeToggle />
        </Stack>
      </Flex>

      <Collapse in={isOpen} animateOpacity>
        <MobileNav />
      </Collapse>
    </Box>
  );
};

<DesktopNav />;
<MobileNav />;
