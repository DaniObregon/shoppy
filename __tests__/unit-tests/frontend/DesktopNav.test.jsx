import "@testing-library/jest-dom";
import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import { DesktopNav } from "../../../client/src/components/main-page/DesktopNav";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import { ChakraProvider } from "@chakra-ui/react";
import { BrowserRouter as Router } from "react-router-dom";
import { useNavigate } from "react-router-dom";

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useNavigate: jest.fn(),
}));

const mockStore = configureStore([]);

describe("DesktopNav component", () => {
  let store;

  // Estado inicial (usuario no autenticado)
  const initialState = {
    userInfo: { role_id: null },
  };

  // Estado cuando el usuario tiene rol 1
  const userRole1State = {
    userInfo: { role_id: 1 },
  };

  // Estado cuando el usuario tiene rol 2
  const userRole2State = {
    userInfo: { role_id: 2 },
  };

  afterEach(() => {
    jest.clearAllMocks();
  });

  test("should render the desktop nav bar with at least one link", () => {
    store = mockStore(userRole1State); // Simular rol 1
    render(
      <Provider store={store}>
        <ChakraProvider>
          <Router>
            <DesktopNav />
          </Router>
        </ChakraProvider>
      </Provider>
    );

    const navLinks = screen.getAllByRole("link");
    expect(navLinks.length).toBeGreaterThan(0);
  });

  test("should not display 'Admin Panel' for roles less than or equal to 1", () => {
    // Probar con rol 1
    store = mockStore(userRole1State);
    render(
      <Provider store={store}>
        <ChakraProvider>
          <Router>
            <DesktopNav />
          </Router>
        </ChakraProvider>
      </Provider>
    );

    expect(
      screen.queryByRole("link", { name: /admin panel/i })
    ).not.toBeInTheDocument(); // Rol 1 no tiene acceso

    // Probar con rol 0 (no autenticado o rol menor)
    store = mockStore(initialState);
    render(
      <Provider store={store}>
        <ChakraProvider>
          <Router>
            <DesktopNav />
          </Router>
        </ChakraProvider>
      </Provider>
    );

    expect(
      screen.queryByRole("link", { name: /admin panel/i })
    ).not.toBeInTheDocument(); // No debería estar presente
  });

  test("should display 'Admin Panel' for roles greater than 1", () => {
    // Probar con rol 2
    store = mockStore(userRole2State);
    render(
      <Provider store={store}>
        <ChakraProvider>
          <Router>
            <DesktopNav />
          </Router>
        </ChakraProvider>
      </Provider>
    );

    expect(
      screen.getByRole("link", { name: /admin panel/i })
    ).toBeInTheDocument(); // Rol 2 tiene acceso
  });

  test("should match snapshot", () => {
    store = mockStore(userRole1State); // Simular rol 1
    const { asFragment } = render(
      <Provider store={store}>
        <ChakraProvider>
          <Router>
            <DesktopNav />
          </Router>
        </ChakraProvider>
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  // ----- 5 - Prueba de Navegación -----
  test("should navigate to the correct path when a link is clicked", () => {
    const mockNavigate = jest.fn();
    useNavigate.mockReturnValue(mockNavigate); // Mock de la función navigate

    store = mockStore(userRole1State); // Simular rol 1
    render(
      <Provider store={store}>
        <ChakraProvider>
          <Router>
            <DesktopNav />
          </Router>
        </ChakraProvider>
      </Provider>
    );

    // Obtener cualquier enlace disponible (ya que no importa cuál en particular)
    const link = screen.getAllByRole("link")[0]; // Obtener el primer enlace para la prueba
    fireEvent.click(link); // Simular clic en el enlace

    // Verificar que mockNavigate haya sido llamado con algún path (puedes ajustarlo si necesitas mayor especificidad)
    expect(mockNavigate).toHaveBeenCalledTimes(1);
    expect(mockNavigate).toHaveBeenCalledWith(expect.any(String)); // Acepta cualquier string como argumento de la función
  });
});
