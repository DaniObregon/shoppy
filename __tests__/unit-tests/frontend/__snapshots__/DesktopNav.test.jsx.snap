// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DesktopNav component should match snapshot 1`] = `
<DocumentFragment>
  <div
    aria-label="desktop navigation"
    class="chakra-stack css-d9vnmu"
  >
    <div
      class="css-0"
    >
      <a
        aria-controls="popover-content-:r1b:"
        aria-expanded="false"
        aria-haspopup="dialog"
        class="chakra-link css-1gbk00m"
        id="popover-trigger-:r1b:"
        role="link"
      >
        Capilares
      </a>
      <div
        class="chakra-popover__popper css-1qq679y"
        style="visibility: hidden; position: absolute; min-width: max-content; inset: 0 auto auto 0;"
      >
        <section
          class="chakra-popover__content css-r1kswd"
          id="popover-content-:r1b:"
          role="tooltip"
          style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.95);"
          tabindex="-1"
        >
          <div
            class="chakra-stack css-8g8ihq"
          >
            <a
              class="chakra-link css-1vqjwd2"
              role="group"
            >
              <div
                class="chakra-stack css-1igwmid"
              >
                <div
                  class="css-0"
                >
                  <p
                    class="chakra-text css-sc37ud"
                  >
                    Coloración
                  </p>
                  <p
                    class="chakra-text css-itvw0n"
                  />
                </div>
                <div
                  class="css-2gz105"
                >
                  <svg
                    class="chakra-icon chakra-icon css-1ossod7"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                      fill="currentColor"
                    />
                  </svg>
                </div>
              </div>
            </a>
            <a
              class="chakra-link css-1vqjwd2"
              role="group"
            >
              <div
                class="chakra-stack css-1igwmid"
              >
                <div
                  class="css-0"
                >
                  <p
                    class="chakra-text css-sc37ud"
                  >
                    Styling
                  </p>
                  <p
                    class="chakra-text css-itvw0n"
                  />
                </div>
                <div
                  class="css-2gz105"
                >
                  <svg
                    class="chakra-icon chakra-icon css-1ossod7"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                      fill="currentColor"
                    />
                  </svg>
                </div>
              </div>
            </a>
            <a
              class="chakra-link css-1vqjwd2"
              role="group"
            >
              <div
                class="chakra-stack css-1igwmid"
              >
                <div
                  class="css-0"
                >
                  <p
                    class="chakra-text css-sc37ud"
                  >
                    HairCare
                  </p>
                  <p
                    class="chakra-text css-itvw0n"
                  />
                </div>
                <div
                  class="css-2gz105"
                >
                  <svg
                    class="chakra-icon chakra-icon css-1ossod7"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                      fill="currentColor"
                    />
                  </svg>
                </div>
              </div>
            </a>
            <a
              class="chakra-link css-1vqjwd2"
              role="group"
            >
              <div
                class="chakra-stack css-1igwmid"
              >
                <div
                  class="css-0"
                >
                  <p
                    class="chakra-text css-sc37ud"
                  >
                    Forma
                  </p>
                  <p
                    class="chakra-text css-itvw0n"
                  />
                </div>
                <div
                  class="css-2gz105"
                >
                  <svg
                    class="chakra-icon chakra-icon css-1ossod7"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                      fill="currentColor"
                    />
                  </svg>
                </div>
              </div>
            </a>
          </div>
        </section>
      </div>
    </div>
    <div
      class="css-0"
    >
      <a
        aria-controls="popover-content-:r1d:"
        aria-expanded="false"
        aria-haspopup="dialog"
        class="chakra-link css-1gbk00m"
        id="popover-trigger-:r1d:"
        role="link"
      >
        Manos y Uñas
      </a>
      <div
        class="chakra-popover__popper css-1qq679y"
        style="visibility: hidden; position: absolute; min-width: max-content; inset: 0 auto auto 0;"
      >
        <section
          class="chakra-popover__content css-r1kswd"
          id="popover-content-:r1d:"
          role="tooltip"
          style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.95);"
          tabindex="-1"
        >
          <div
            class="chakra-stack css-8g8ihq"
          >
            <a
              class="chakra-link css-1vqjwd2"
              role="group"
            >
              <div
                class="chakra-stack css-1igwmid"
              >
                <div
                  class="css-0"
                >
                  <p
                    class="chakra-text css-sc37ud"
                  >
                    Prueba children
                  </p>
                  <p
                    class="chakra-text css-itvw0n"
                  />
                </div>
                <div
                  class="css-2gz105"
                >
                  <svg
                    class="chakra-icon chakra-icon css-1ossod7"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                      fill="currentColor"
                    />
                  </svg>
                </div>
              </div>
            </a>
          </div>
        </section>
      </div>
    </div>
    <div
      class="css-0"
    >
      <a
        aria-controls="popover-content-:r1f:"
        aria-expanded="false"
        aria-haspopup="dialog"
        class="chakra-link css-1gbk00m"
        id="popover-trigger-:r1f:"
        role="link"
      >
        Maquillaje
      </a>
    </div>
    <div
      class="css-0"
    >
      <a
        aria-controls="popover-content-:r1h:"
        aria-expanded="false"
        aria-haspopup="dialog"
        class="chakra-link css-1gbk00m"
        id="popover-trigger-:r1h:"
        role="link"
      >
        Equipamientos
      </a>
    </div>
    <div
      class="css-0"
    >
      <a
        aria-controls="popover-content-:r1j:"
        aria-expanded="false"
        aria-haspopup="dialog"
        class="chakra-link css-1gbk00m"
        id="popover-trigger-:r1j:"
        role="link"
      >
        Estética y Spa
      </a>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
